src/Middleware/HeaderPropagation/test/HeaderPropagationMiddlewareTest.cs

```diff
@@ -185,43 +185,5 @@ public async Task MultipleEntries_AddsFirstToProduceValue()
             Assert.Contains("in", CapturedHeaders.Keys);
             Assert.Equal("Test", CapturedHeaders["in"]);
         }
-
-        [Fact]
-        public async Task HeaderInRequest_WithBleedAsyncLocal_HasCorrectValue()
```

Comment:
I removed these tests because it's the responsibility of the server, not the middleware, to ensure it has a fresh ExecutionContext for each request. The HeaderPropagationMiddleware does not have any functional tests, and @benaadams [already added functional tests](https://github.com/dotnet/aspnetcore/pull/14146/files#diff-d47b9baa7ec5338570136a89e5e8a4deR288) to verify IAsyncLocals in general work as expected with Kestrel.

That said, I did manually verify that #14146 fixes things so that the HeaderPropagationMiddleware sees a fresh HeaderPropagationValues each request after removing `async` from Invoke. And verified this was not the case without the changes from #14146.